legacy/firmware/config.c

```diff
@@ -406,27 +407,36 @@ void config_init(void) {
 }
 
 void session_clear(bool lock) {
-  sessionSeedCached = secfalse;
-  memzero(&sessionSeed, sizeof(sessionSeed));
-  sessionIdCached = secfalse;
-  memzero(&sessionId, sizeof(sessionId));
+  for (uint8_t i = 0; i < MAX_SESSIONS_COUNT; i++) {
+    session_clearCache(sessionsCache + i);
+  }
+  activeSessionCache = NULL;
   if (lock) {
-    storage_lock();
+    session_lock();
   }
 }
 
+void session_clearCache(Session *session) {
+  session->usage = 0;
+  memzero(session->id, sizeof(session->id));
+  memzero(session->seed, sizeof(session->seed));
+  session->seedCached = false;
+}
+
+void session_lock(void) { storage_lock(); }
```

Comment:
I don't see the motivation behind this.

Reply:
It's basically just a syntax sugar. `session_lock` is called in places where you work with sessions, however not with storage directly. Also, it seemed reasonable to provide a function that locks the session, even though it is currently implemented as locking the storage only. It can be extended easily in future.

Reply:
1. I don't like wrapper functions that don't add any functionality, because they just obfuscate the code. Of course, if we know we will be adding some functionality here in the future, then that's perfectly fine. I just couldn't to think of anything that would make sense here.
2. The name doesn't make much sense to me. It doesn't lock a session. It locks the device, i.e. all sessions and possibly more.

Reply:
I mean, I feel this does not have anything to do with sessions and I just find it misleading.

... I see this is related to the naming of the protobuf messages and https://github.com/trezor/trezor-firmware/issues/819. So if we really need this function (which I doubt), then let's call it `lock_device()` or something.

Reply:
1. I agree that wrapper functions are not usually needed. But I believe in this case there is a real benefit. By providing a function in `config.c` to lock the device, we provide a clear interface to perform such action. The fact that it is currently only `storage_lock` is not important. A new developer/3rd party will not realize `storage_lock == lock_device`.
2. I agree the name is not good and I agree with `lock_device`.

Reply:
Well, actually this should be `config_lockDevice` to fit the others.

Reply:
`config_lockDevice()` is perfect.